numpy/ma/core.py

```diff
@@ -4563,48 +4584,41 @@ def all(self, axis=None, out=None):
         """
         mask = _check_mask_axis(self._mask, axis)
         if out is None:
-            d = self.filled(True).all(axis=axis).view(type(self))
+            d = self.filled(True).all(axis=axis, **kwargs).view(type(self))
             if d.ndim:
                 d.__setmask__(mask)
-            elif mask:
+            elif mask and self.size > 0:
```

Comment:
What is this check for? if it is there, ndim should be 0, no? Or was this the problem you mentioned in this other PR?